flipperdevices/flipperzero-firmware
# [FL-1268] Gauge re-calibration

# What's new

- Change battery profile with 0 capacity at 3.3 cell voltage point
- Add gauging configuration with adjusted smoothing parameters
- Set battery profile only if it differs from the one that gauge use
- Change poweroff condition. The poweroff countdown starts when battery capacity is zero and no voltage on VBUS detected

# Verification 

- Build for f5 and f6 targets
- Reset gauge by removing battery. Check in logs that gauge update battery profile.
- Verify correct Design Capacity and Full Charge Capacity with power_test CLI command
- Reset flipper. Check in logs that gauge skips battery profile update
- Discharge battery. Verify that Remain Capacity approaches zero while Gauge Voltage approaches 3.3V
- Verify poweroff countdown starting when Remain Capacity is 0

# Checklist (do not modify)
